### ISSUE TITLE
Discrepancies when linting through STDIN

### ISSUE THREAD

**Issue by jamieconnolly** [brigade/scss-lint#799]
Bear with me, it's a pretty long post 😝!

So, given that I have a file `css/foo.scss` that contains the following (`css/bar.scss` is empty):

```sass
---
---
@import 'bar';

* {
  background-color: red !important;
}
```

And an `.scss-lint.yml` file that looks like this:

```yml
preprocess_command: "sed '1,2s/---//'"
preprocess_files:
  - css/foo.scss
linters:
  ColorKeyword:
    enabled: true
  ColorVariable:
    enabled: false
  ImportantRule:
    enabled: true
    exclude:
      - css/foo.scss
```

If I run, `scss-lint` from the base directory, I get the following output (exactly as expected):

```shellsession
$ scss-lint --config=.scss-lint.yml
css/foo.scss:7 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
```

If I then run `scss-lint` from inside the `css` directory, I get the following output (again, as expected):

```shellsession
$ scss-lint --config=../.scss-lint.yml
foo.scss:7 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
```

Now changing it up to pass through the file contents to `scss-lint` via `STDIN`, if I run the following command from the base directory, I get this output (as expected):

```shellsession
$ cat css/foo.scss | scss-lint --config=.scss-lint.yml --stdin-file-path=css/foo.scss
css/foo.scss:7 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
```

If I run the following command from inside the `css` directory, passing through the `--stdin-file-path` relative to the `.scss-lint.yml` file, I get this output:

```shellsession
$ cat foo.scss | scss-lint --config=../.scss-lint.yml --stdin-file-path=css/foo.scss
css/foo.scss:7 [W] ImportantRule: !important should not be used
css/foo.scss:7 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
```
As you can see, it doesn't handle the `exclude` parameter of the linter.

If I then run a similar command, again from inside the `css` directory, but this time setting the `--stdin-file-path` to relative of the current directory, I get this output (the output has been manually concatenated):

```shellsession
$ cat foo.scss | scss-lint --config=../.scss-lint.yml --stdin-file-path=foo.scss
foo.scss:4 [E] Syntax Error: Invalid CSS after "@import 'bar'": expected "{", was ";"
foo.scss:7 [W] ColorKeyword: Color `red` should be written in hexadecimal form as `#ff0000`
```
As you can see, it handles the `exclude` parameter of the linter, but doesn't handle the `preprocess_files` option. This is the opposite of the previous command.

I'm not sure which is the preferred way, but surely both `exclude` and `preprocess_files` should work the same way?

Also, in case it matters, only the individual linter's `exclude` parameter is handled via `STDIN`, the "exclude all" parameter does not.

Thanks for reading!!!

**Comment by sds** [comment]
Thanks for the details explanation of the issue, @jamieconnolly.

Would love to get a pull request that addresses this, as it's a pretty specific issue that would take some time to fix but would only benefit a small cohort (I could be wrong). I also (personally) don't have time available to support niche issues such as this. 

Again, happy to merge a pull request! Thanks so much for the report.
